relay/blob_provider.go

```diff
@@ -20,6 +21,9 @@ type blobProvider struct {
 
 	// blobCache is an LRU cache of blobs.
 	blobCache cache.CachedAccessor[v2.BlobKey, []byte]
+
```

Comment:
I think this actually needs to diverge from the normal patterns a bit.

The complicating factor is that we want to prevent two or more GRPCs that request the same blob from all doing an S3 fetch. This is the core purpose of the `CachedAccessor` class -- it forces duplicate requests to wait if there is already a fetch in progress. If we used the context from one of the callers, but then that connection broke and the context got cancelled, then other callers waiting on that data would instead get an error. So instead, the internal fetch happens using the server's context which intentionally ignores if the original caller cancels its context.

Reply:
> The problem with using a global ctx is that this whole component is dead when that context gets killed

Yes. The context passed into the constructor of the provider should survive as long as the relay is running. Let's discuss this in our 1-1 today.